store/sqlstore/user_store.go

```diff
@@ -1061,25 +1053,29 @@ func (us SqlUserStore) GetSystemAdminProfiles() (map[string]*model.User, *model.
 	return userMap, nil
 }
 
-func (us SqlUserStore) GetByEmail(email string) (*model.User, *model.AppError) {
+func (us SqlUserStore) GetByEmail(email string) (*model.User, error) {
 	query := us.usersQuery.Where("Email = lower(?)", email)
 
 	queryString, args, err := query.ToSql()
 	if err != nil {
-		return nil, model.NewAppError("SqlUserStore.GetByEmail", "store.sql_user.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "get_by_email_tosql")
 	}
 
 	user := model.User{}
 	if err := us.GetReplica().SelectOne(&user, queryString, args...); err != nil {
-		return nil, model.NewAppError("SqlUserStore.GetByEmail", store.MISSING_ACCOUNT_ERROR, nil, "email="+email+", "+err.Error(), http.StatusInternalServerError)
+		if err == sql.ErrNoRows {
```

Comment:
Let's keep the logic same here.

Reply:
I have some thoughts here so let's leave this 2 suggestions to the end

Reply:
I didn't know whether to change it or not, because the client code (`app/user.go`) was already prepared to handle a 404 but by comparing the key of the message (using string comparison to determinate if that was an 404 or not), so I think I'm not changing the behaviour here because we are doing the same but using store errors instead (which I think is better than string comparison)

Reply:
I do understand that, but I am just burnt by things breaking in the past, where we missed a case that assumed 500 instead of 404. And I just don't want to repeat that. Unless there's a good reason to do this, I'd be prefer not to change this.

Reply:
Yep I understand the point, but in this case if I'm understanding well if I remove that condition I will change the behaviour